src/framework/audio/audiomodule.cpp

```diff
@@ -258,14 +271,21 @@ void AudioModule::onDestroy()
 
 void AudioModule::setupAudioDriver(const IApplication::RunMode& mode)
 {
+    const bool shouldMeasureInputLag = m_configuration->shouldMeasureInputLag();
+
     IAudioDriver::Spec requiredSpec;
     requiredSpec.sampleRate = m_configuration->sampleRate();
     requiredSpec.format = IAudioDriver::Format::AudioF32;
     requiredSpec.channels = m_configuration->audioChannelsCount();
     requiredSpec.samples = m_configuration->driverBufferSize();
-    requiredSpec.callback = [this](void* /*userdata*/, uint8_t* stream, int byteCount) {
+    requiredSpec.callback = [this, shouldMeasureInputLag](void* /*userdata*/, uint8_t* stream, int byteCount) {
         auto samplesPerChannel = byteCount / (2 * sizeof(float));
-        m_audioBuffer->pop(reinterpret_cast<float*>(stream), samplesPerChannel);
+        float* dest = reinterpret_cast<float*>(stream);
+        m_audioBuffer->pop(dest, samplesPerChannel);
+
+        if (shouldMeasureInputLag) {
```

Comment:
Nitpick: 
Isn't this callback called very often? If so, that means that this bool is checked very often... you could optimise it by having two callbacks, one with measuring and one without, and assign the desired callback based on the bool. I.e. check the bool once, outside the callback.